spec/latest/json-ld-framing/index.html

```diff
@@ -552,34 +543,109 @@
-      property</a>, and <em>value</em> for <em>element</em>.</li>
-    <li>Skip step <strong>12</strong>: <a>free-floating nodes</a> are retained.</li>
-  </ul>
+  <p>The Frame Matching Algorithm is used as part of the <a href="#framing-algorithm">Framing algorithm</a>
+    to determine if a particular <a>node object</a> matches the critera set in a <a>frame</a>.
+    In general, a node object matches a frame if it meets the matches on <code>@type</code>,
+    or <code>@id</code>,
+    or if it matches given one of several different properties (or all properties, if the
+    <a>require all flag</a> is present.).</p>
+
+  <p class="note">As matching is performed on expanded node objects, all values will be in the form of an array.</p>
+
+  <p>Node matching uses a combination of JSON constructs to match <em>any</em>, <em>no</em>, or <em>some</em> specific values:</p>
+  <dl>
+    <dt><code>[]</code> (<code><dfn>match none</dfn></code>)</dt>
+    <dd>An empty array matches no values, or a value which is, itself, an empty array.</dd>
+    <dt><code>{}</code> (<code><dfn>wildcard</dfn></code>)</dt>
+    <dd>An array containing an empty object
+      (after excluding the framing properties <code>@default</code>, <code>@embed</code>, <code>@explicit</code>, <code>@omitDefault</code>, and <code>@requireAll</code>)
+      matches any value that is present, and does not match if there are no values.</dd>
+    <dt><code>[<a>IRI</a>+]</code></dt>
+    <dd>One or more strings in the form of an IRI, used for matching on <code>@type</code> and <code>@id</code>.</dd>
+    <dt><code>[<a>node object</a>]</code> (<code><dfn>node pattern</dfn></code>)</dt>
+    <dd>A non-empty <a>node object</a>, used to match specific values using recursive node matching.</dd>
+    <dt><code>[<a>value object</a>]</code> (<code><dfn>value pattern</dfn></code>)</dt>
+    <dd>A <a>value object</a>, used to match a specific value. Within a <a>value object</a>,
+      the values for <code>@value</code>, <code>@type</code>, and <code>@language</code>
+      may also be an array of one or more <a>string</a> values.</dd>
+  </dl>
+
+<p>The frame matching algorithm takes the <a>framing state</a> (<em>state</em>),
+  a list of subjects to match from the <a>map of flattened subjects</a> (<em>subjects</em>),
+  a <a>frame</a> to match against (<em>frame</em>), and the <em>requireAll</em> flag
+  and returns a list of matched subjects by filtering each <em>node</em> in <em>subjects</em> as follows:</p>
+
+<p>Frame matching follows a cascade, first attempting to match on a particular <code>@id</code>, then
+  a particular <code>@type</code> (or lack of <code>@type</code>), then by matching on any or all
+  of a set of properties, if neither <code>@id</code>, nor <code>@type</code> are in the frame.</p>
+
+<ol class="algorithm">
+  <li><em>Node</em> matches if it has an <code>@id</code> property including any <a>IRI</a> in the <code>@id</code> property in <em>frame</em>.</li>
```

Comment:
Note: it may not be clear to people that: `"@id": []` won't work for capturing blank nodes ... because the flattened subject map will assign a blank node to everything (IIRC).